### PR TITLE
Document feature flag process

### PR DESCRIPTION
**Pull request by jryans**
This records the feature flag process we intend to use with Riot and also how
that interacts with other teams and configuration.

Fixes https://github.com/vector-im/riot-web/issues/11116

### CODE REVIEW

File: docs/config.md [hunk 1]
```diff
@@ -22,9 +22,10 @@ For a good example, see https://riot.im/develop/config.json.
      `default_hs_url` is specified. When multiple sources are specified, it is unclear
      which should take priority and therefore the application cannot continue.
    * As of Riot 1.4.0, identity servers are optional. See [Identity servers](#identity-servers) below.
-1. `features`: Lookup of optional features that may be `enable`d, `disable`d, or exposed to the user
-   in the `labs` section of settings.  The available optional experimental features vary from
-   release to release. The available features are described in [labs.md](labs.md).
+1. `features`: Lookup of optional features that may be `enable`d, `disable`d, or
+   exposed to the user in the `labs` section of settings. The available
+   optional experimental features vary from release to release and are (usually) [documented](labs.md). The feature flag process is
```
Comment (turt2live) [inline docs/config.md]: we should be better:

```suggestion
   optional experimental features vary from release to release and are [documented](labs.md). The feature flag process is
```

File: docs/config.md [hunk 2]
```diff
@@ -22,9 +22,10 @@ For a good example, see https://riot.im/develop/config.json.
...
+   [documented](feature-flags.md) as well.
```
Comment (turt2live) [inline docs/config.md]:
```suggestion
   is [documented](feature-flags.md) as well.
```

File: docs/feature-flags.md [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+Under the [new spec
+process](https://github.com/matrix-org/matrix-doc/pull/2324), server-side
+unstable features can be used by clients and enabled by default as long as
+clients commit to doing the associated clean up work once a feature stabilises.
+
+## Starting work on a feature
+
+When starting work on a feature, we should create a matching feature flag:
+
+* Add a new
```
Comment (turt2live) [inline docs/feature-flags.md]: numbered?

File: docs/feature-flags.md [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+
+With these steps completed, the feature is disabled by default, but can be
+enabled on develop by interested users for testing.
+
+Different features may have different deployment plans for when to enable where. The
+following lists a few common options.
+
+## Enabling by default on develop
+
+Set the feature to `enable` in the [develop config](../riot.im/develop/config.json):
```
Comment (turt2live) [inline docs/feature-flags.md]: a github link would be great for those who are reading outside of github.

applies throughout PR.

File: docs/feature-flags.md [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+
+## Enabling by default on staging and app
+
+Set the feature to `enable` in the [app config](../riot.im/app/config.json).
+
+## Feature deployed successfully
+
+Once we're confident that a feature is working well, we should remove the flag:
+
+* Remove the [setting](https://github.com/matrix-org/matrix-react-sdk/blob/develop/src/settings/Settings.js)
```
Comment (turt2live) [inline docs/feature-flags.md]: there is potentially a stage for converting it to a real setting

File: docs/config.md (hunk 1 above)
Comment (jryans) [reply docs/config.md]: Haha, okay, fair enough. 😄 At least the linked to doc explains it may not be exhaustive...

File: docs/config.md (hunk 2 above)
Comment (jryans) [reply docs/config.md]: This already exists on the previous line.

File: docs/feature-flags.md (hunk 3 above)
Comment (jryans) [reply docs/feature-flags.md]: Sure, makes sense.

File: docs/feature-flags.md (hunk 4 above)
Comment (jryans) [reply docs/feature-flags.md]: Okay, done. (I was very tempted to convert to reference style, but I guess that would still confuse...)

File: docs/feature-flags.md (hunk 5 above)
Comment (jryans) [reply docs/feature-flags.md]: Agreed, I have added that as well.

**Pull request merged**